src/Talis/Persona/Client/Tokens.php

```diff
@@ -13,14 +13,15 @@ class Tokens extends Base
      * Validates the supplied token using JWT or a remote Persona server.
      * An optional scope can be supplied to validate against. If a token
      * is not provided within the parameter one will be extracted from
-     * either $_SERVER, $_GET or $_POST.
+     * either $_SERVER, $_GET or $_POST. If scope parameter is an array and at
+     * least one of the scopes can be validated, the result is a success.
      *
      * The order of validation is as follows: JWT, local Redis cache, then remote Persona.
      *
      * @param array $params a set of optional parameters you can pass to this method <pre>
      *      access_token: (string) a token to validate explicitly, if you do not specify one the method tries to find one,
      *      scope: (string|array) specify this if you wish to validate a scoped token
-     * @return bool|string will return false if could not validate the token, else true
+     * @return int|boolean true on success else ValidationErrors enum
```

Comment:
This method can return either a boolean or an integer! I'm not keen on that although I note it used to return a boolean or a string.

I think it would be better if instead of ValidationErrors the enum was ValidationResults one of which was success.

Seeing as we were already mixing return types on this method, I wouldn't want to block this if it needs to go down. But changing this would be good in my opinion.